src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.cs

```diff
@@ -948,15 +988,52 @@ public SolutionCompilationState WithAnalyzerConfigDocumentText(
     }
 
     /// <inheritdoc cref="Solution.WithDocumentSyntaxRoots(ImmutableArray{ValueTuple{DocumentId, SyntaxNode}}, PreservationMode)"/>
+    /// <remarks>
+    /// This method supports updating ordinary documents and source generated documents.
+    /// </remarks>
     public SolutionCompilationState WithDocumentSyntaxRoots(ImmutableArray<(DocumentId documentId, SyntaxNode root)> syntaxRoots, PreservationMode mode)
```

Comment:
intresting. lots of duping between this method and the one above.  not a requiremetn for this pr, but i think we could make this work through a helper.  i'm willing to help with that one this PR goes in.